### PR TITLE
Add automatic HSTS for SSL-capable sites

### PR DESCRIPTION
**Pull request by rmccue**
This adds automatic HSTS for sites when served over SSL, while allowing it to be overridden or disabled (as documented).

### CODE REVIEW

**Comment by rmccue** [comment]
This should be released as a new major version, as it's significant new behaviour.

**Comment by rmccue** [comment]
Alternatively, I can disable the auto-behaviour for a minor release, and change that only in a major release as a follow-up?

**Comment by shadyvb** [comment]
Yeah, that makes sense, so we can activate wherever without needing to wait for the next major release.

**Comment by shadyvb** [comment]
Or maybe even a version-conditional default ?

**Review by shadyvb — APPROVED** [review]
LGTM 👍 Minus the default value adjustment for the minor release.

**Comment by rmccue** [comment]
Split the default to #18; just waiting for tests for complete before I merge.

**Review by shadyvb — APPROVED** [review]

**Comment by roborourke** [comment]
Where's the issue for this? Wondering if we need to look at this for nginx / cloudfront rules too.

**Pull request merged**
